Bump vlc_telnet to 2.0.1, several bug fixes

dmcc opened the pull request:
Switch to python-telnet-vlc 2.0.1 and avoid some unnecessary int/str
conversions.

Add `SUPPORT_VOLUME_STEP`.

Fix various state issues (initial state not sent as part of first
update, unmute detection, seeking didn't actually work).

<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->


## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->


## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [x] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example entry for `configuration.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example configuration.yaml

```

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [x] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

dmcc committed:
vlc_telnet: Switch to 2.0.1, several bug fixes

Switch to python-telnet-vlc 2.0.1 and avoid some unnecessary int/str
conversions.

Add SUPPORT_VOLUME_STEP.

Fix various state issues (initial state not sent as part of first
update, unmute detection, seeking didn't actually work).

homeassistant commented:
Hi @dmcc, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#44776).

Once you do that we will be able to review and accept this pull request.

Thanks!

MartinHjelmare commented:
Please split this PR into at least three PRs:

- Bump library version
- Fix issues
- Add volumestep support

ludeeus reviewed (changes requested):
> Please split this PR into at least three PRs:
> 
> * Bump library version
> * Fix issues
> * Add volume step support

Marking this as "Request changes"

dmcc commented:
Sorry for the delay, finally had cycles to test what happens if we slice it up as requested. Unfortunately, while `python-telnet-vlc` is a general improvement, Home Assistant relies on bad its APIs. So, if we just bump the `python-telnet-vlc` version, we'll actually introduce two new issues (volume control will break and we'll no longer get any position status info).

Which course of action would you like to pursue?

1. Split into 3+ PRs as requested (this is presumably fine as long as they're not split across Home Assistant releases)
2. Split into 2 PRs: (1) bump `python-telnet-vlc` version and address breaking changes + fix bugs (mute/unmute detection, fix initial state update), (2)  add `SUPPORT_VOLUME_STEP`
3. Split into 3 PRs: (1) bump `python-telnet-vlc` version and address breaking changes, (2) separate bugfixes (mute/unmute detection, fix initial state update), (3)  add `SUPPORT_VOLUME_STEP`
4. Keep this as one larger PR
5. (something else?)

MartinHjelmare commented:
Please do 3.

dmcc commented:
Alright, I've extracted the first of several PRs into https://github.com/home-assistant/core/pull/46608 (apologies if there's a better way to do this in git).

frenck commented:
Seems like 3 PR are made and merged. As far as I can see, the update handling changes have not ended up in the other PRs.

Is there still a PR on route? Or are you planning on rebasing/updating this PR?

dmcc commented:
Hey @frenck, thanks for checking in. Right, update handling changes still haven't been pulled out and I'm still planning on getting to this (there might be others too). That said, there's probably no point in keeping this one open since it's not going anywhere, so will close it.
